src/components/AttachmentCarousel/createInitialState.js

```diff
@@ -49,14 +48,16 @@ function createInitialState(props) {
         attachments.reverse();
     }
 
-    const page = _.findIndex(attachments, (a) => a.source === props.source);
-    if (page === -1) {
-        Navigation.dismissModal();
-        return;
+    const source = tryResolveUrlFromApiRoot(props.source);
```

Comment:
I think we're duplicating the code here. We're using the same method in `AttachmentCarousel/index.js`.

Reply:
We can call it once here `source={tryResolveUrlFromApiRoot(props.source)}`
https://github.com/Expensify/App/blob/4700f96d5abb98d40f2b30a2385d296bbe45de90/src/components/AttachmentModal.js#L345

Reply:
Let's do that.